deepspeed/__init__.py

```diff
@@ -66,6 +66,50 @@ def _parse_version(version_str):
 dist = None
 
 
+def _mark_initialized(trainobj: Union[torch.nn.Module, Optimizer, _LRScheduler]):
+    """Mark a trainobj as initialized by setting the ds_is_inited attribute to True."""
+    if hasattr(trainobj, 'ds_is_inited'):
+        assert trainobj.ds_is_inited, "Not expecting the training object has `ds_is_inited` to be False if it exists, make sure you didn't set it to False or called deepspeed.initialize on the model more than once."
+        return
+
+    trainobj.ds_is_inited = True
+
+
+def _is_initialized(trainobj: Union[torch.nn.Module, Optimizer, _LRScheduler]):
```

Comment:
Make name a bit deepspeed-specific to avoid collisions
```suggestion
def _is_ds_initialized(trainobj: Union[torch.nn.Module, Optimizer, _LRScheduler]):
```